### PR TITLE
Fix incorrect column and escaped colons for `selector-pseudo-element-colon-notation` rule

### LINKED ISSUE #3310: Fix false positives for escaped colons in selector-pseudo-element-colon-notation
**Describe the bug? (A clear and concise description of what the bug is)**

Stylelint expects a class named `.pseudo\:before-none` to have two colons even if the colon in the name is escaped and is not targeting any pseudo elements.

I'm using this _weird_ naming convention to make my classes match Tailwind's.

**Which rule, if any, is this issue related to?**

`selector-pseudo-element-colon-notation`

**What CSS is needed to reproduce this issue?**

```css
.pseudo\:before-none::before { content: none; }
.pseudo\:after-none::after { content: none; }
```

**What stylelint configuration is needed to reproduce this issue?**

```json
{
  "rules": {
    "selector-pseudo-element-colon-notation": "double",
  }
}
```

**Which version of stylelint are you using?**

`9.2.0`

**How are you running stylelint: CLI, PostCSS plugin, Node API?**

Via [vscode-stylelint](https://github.com/shinnn/vscode-stylelint)

**Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?**

No.

**What did you expect to happen?**

No warnings to be flagged when the colon in `:before` or `:after` is escaped.

**What actually happened (e.g. what warnings or errors you are getting)?**

The following warnings were flagged:

```shell
Expected double colon pseudo-element notation (selector-pseudo-element-colon-notation)
```

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

- Closes #3310
- Part of #5499

> Is there anything in the PR that needs further explanation?

Replacing `style-search` with `postcss-selector-parser`.

### CODE REVIEW

File: lib/utils/parseSelector.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const selectorParser = require('postcss-selector-parser');
 module.exports = function parseSelector(selector, result, node, callback) {
 	try {
 		return selectorParser(callback).processSync(selector);
-	} catch {
-		result.warn('Cannot parse selector', { node, stylelintType: 'parseError' });
+	} catch (err) {
+		result.warn(`Cannot parse selector (${err})`, { node, stylelintType: 'parseError' });
```
Comment (ybiquitous) [inline lib/utils/parseSelector.js:16]: [note] It is hard to investigate a raised error in `callback` unless `err`.

**Review by jeddy3 — APPROVED** [review]
Great to see this bug fixed, and much cleaner code using the parser.

LGTM!

**Review by jeddy3 — APPROVED** [review]

**Comment by ybiquitous** [comment]
@jeddy3 Oh, thank you for your quick review!
Yes, the `postcss-selector-parser` API is so easier to use. 👍🏼

**Comment by ybiquitous** [comment]
@jeddy3 Can we merge this PR? Or, are more approvals needed?

**Comment by jeddy3** [comment]
I think we can merge refactoring pull requests with one review.

**Comment by ybiquitous** [comment]
I got it. 👍🏼  Updated changelog:

> Fixed: `selector-pseudo-element-colon-notation` false positives for escaped colons ([#5879](https://github.com/stylelint/stylelint/pull/5879)).

**Pull request merged**